src/jdk.crypto.ec/share/classes/sun/security/ec/ECPrivateKeyImpl.java

```diff
@@ -195,4 +201,20 @@ private void parseKeyBits() throws InvalidKeyException {
             throw new InvalidKeyException("Invalid EC private key", e);
         }
     }
+
+    @Override
+    public PublicKey calculatePublicKey() {
+        ECParameterSpec ecParams = getParams();
+        ECOperations ops = ECOperations.forParameters(ecParams)
+                .orElseThrow(ProviderException::new);
+        MutablePoint pub = ops.multiply(ecParams.getGenerator(), getArrayS0());
+        AffinePoint affPub = pub.asAffine();
+        ECPoint w = new ECPoint(affPub.getX().asBigInteger(),
+                affPub.getY().asBigInteger());
+        try {
+            return new ECPublicKeyImpl(w, ecParams);
+        } catch (InvalidKeyException e) {
+            throw new ProviderException("Should not happen", e);
```

Comment:
On the remote chance that this exception is thrown, the message doesn't help the user.  A better message might be "Unexpected error creating public key" or even just not having a message and using the `ProviderException(Throwable)`constructor. 

This message is also used in XDHPrivateKeyimpl.